src/js/components/devices/devices.js

```diff
@@ -54,26 +84,36 @@ export class Devices extends React.Component {
   }
 
   componentDidMount() {
-    clearAllRetryTimers(this.props.setSnackbar);
+    const { filteringAttributes, getAllDeviceCounts, match, selectGroup, setDeviceFilters, setSnackbar } = this.props;
+    clearAllRetryTimers(setSnackbar);
     this._restartInterval();
-    this.props.getAllDeviceCounts();
-    if (this.props.match.params.filters) {
-      let groupName = '';
-      const str = decodeURIComponent(this.props.match.params.filters);
-      const filters = str.split('&').reduce((filters, filter) => {
-        const filterPair = filter.split('=');
-        if (filterPair[0] === 'group') {
-          groupName = filterPair[1];
-        } else {
-          const scope = filterPair[0] === 'id' ? { scope: 'identity' } : {};
-          filters.push({ ...emptyFilter, ...scope, key: filterPair[0], value: filterPair[1] });
-        }
-        return filters;
-      }, []);
+    getAllDeviceCounts();
+    if (match.params.filters) {
+      const { filters, groupName } = convertQueryToFilterAndGroup(match.params.filters, filteringAttributes);
+      if (groupName) {
+        selectGroup(groupName, filters);
+      } else {
+        setDeviceFilters(filters);
+      }
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    const { filteringAttributes, location, selectGroup, setDeviceFilters } = this.props;
+    if (
+      location.search &&
+      !Object.values(prevProps.filteringAttributes)
+        .flat()
+        .some(attribute => attribute) &&
+      Object.values(filteringAttributes)
+        .flat()
+        .some(attribute => attribute)
+    ) {
+      const { filters, groupName } = convertQueryToFilterAndGroup(location.search, filteringAttributes);
       if (groupName) {
-        this.props.selectGroup(groupName, filters);
+        selectGroup(groupName, filters);
       } else {
-        this.props.setDeviceFilters(filters);
+        setDeviceFilters(filters);
       }
```

Comment:
I'm not sure I understand why you work on location.search here, instead of location.hash. What I see is that this block is equal to the one in componentDidMount, apart from working on location.search instead of match.params.filters. Maybe we can refactor them and share the implementation across the two different methods? Does it make sense?

Reply:
Good point and I wanted to unify the two, but had to keep the `match.params.filter` one for compatibility with the current link style `/devices/<deviceId>` and the search one for the `/devices?id=<deviceId>` one that is unfortunately unsupported by react-router... I'll refactor to extract more shared code though👌

Reply:
Ah, I see, now I understand. No need to refactor if you think it's useless, the rest of the PR looks good and works fine here.